test/parallel/test-https-immutable-options.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const https = require('https');
+const tls = require('tls');
+
+const dftProtocol = {};
+tls.convertNPNProtocols([ 'http/1.1' ], dftProtocol);
+
+const opts = { foo: 'bar', NPNProtocols: [ 'http/1.1' ] };
+const server1 = https.createServer(opts);
+
+assert.deepStrictEqual(opts, { foo: 'bar', NPNProtocols: [ 'http/1.1' ] });
+assert.strictEqual(server1.NPNProtocols.compare(dftProtocol.NPNProtocols), 0);
+
+const mustNotCall = common.mustNotCall();
+const server2 = https.createServer(mustNotCall);
```

Comment:
The comment below doesn't match the behavior of this line. If we're testing no arguments, it should just be `https.createServer()`.